cosmos-genesis-tinker/cosmos_genesis_tinker.py

```diff
@@ -0,0 +1,349 @@
+                balance["address"] = new_address
+                found_balance = True
+                break
+
+        if not found_balance:
+            raise Exception('Could not find old balance')
+
+        found_delegation = False
+        for delegation in delegations:
+            if delegation["delegator_address"] == old_address:
+                delegation["delegator_address"] = new_address
+                found_delegation = True
+                break
+
+        if not found_delegation:
+            raise Exception("Could not find old delegator stake")
+
+        for info in starting_infos:
+            if info["delegator_address"] == old_address:
+                info["delegator_address"] = new_address
+                break
+
+        return self
+
+    def create_coin(self, denom, amount="0"):
+        """
+        Creats a new coin based on a given name if it doesn't exist
+        """
+
+        supplies = self.genesis["app_state"]["bank"]["supply"]
+
+        for supply in supplies:
+            if supply['denom'] == denom:
+                # Already exists, so we don't need to add it
+                return self
+
+        supplies.append({"denom": denom, "amount": amount})
+
+        return self
+
+    def increase_balance(self, address, increase=300000000, denom="uatom"):
+        """
+        Increases the balance of a person and also the overall supply of uatom
+        """
+
+        balances = self.genesis["app_state"]["bank"]["balances"]
+
+        found_balance = False
+        for balance in balances:
+            if balance["address"] == address:
+                found_balance = True
+                had_coin = False
+                for coin in balance["coins"]:
+                    if coin["denom"] == denom:
+                        old_amount = int(coin["amount"])
+                        new_amount = old_amount + increase
+                        coin["amount"] = str(new_amount)
+                        had_coin = True
+                        break
+                if not had_coin:
+                    balance["coins"].append(
+                        {"denom": denom, "amount": str(increase)})
+                break
+
+        if not found_balance:
+            raise Exception('Could not find balance for address')
+
+        self.increase_supply(increase, denom)
+        return self
+
+    def increase_supply(self, increase, denom="uatom"):
+        """
+        Increase the total supply of coins of a given denomination
+        """
+
+        supplies = self.genesis["app_state"]["bank"]["supply"]
+
+        found_coin = False
+        for coin in supplies:
+            if coin["denom"] == denom:
+                old_amount = int(coin["amount"])
+                new_amount = old_amount + increase
+                coin["amount"] = str(new_amount)
+                found_coin = True
+                break
+
+        if not found_coin:
+            self.create_coin(denom, increase)
+
+        return self
+
```

Comment:
There are a few linked changes, so I think they should be wrapped in an API, something like:

```suggestion
def increase_delegator_stake_to_validator(self, delegator, validator, stake, token_bonding_pool_address=TOKEN_BONDING_POOL_ADDRESS):
  self.increase_balance(token_bonding_pool_address, stake)
  self.increase_delegator_stake(delegator, stake)
  self.increase_validator_stake(validator, stake)
  self.increase_validator_power(validator, stake)
```